torch/_inductor/comm_lowering.py

```diff
@@ -114,10 +113,12 @@ def realize_as_comm_buffer(
 def _get_data(x: ir.TensorBox) -> ir.IRNode:
     if isinstance(x.data, ir.BaseView):
         # TensorBox -> *View -> StorageBox -> IRNode
-        return x.data.unwrap_view().data
+        node = x.data.unwrap_view()
+        assert isinstance(node, (ir.BaseView, ir.MutableBox))
```

Comment:
This isn't a null check, right?

Reply:
Often a good guess, but in this case, `node` has deduced type `IRNode`, and only `BaseView` and `MutableBox` have a `.data` field with type `IRNode`.

This specific check is repeated a lot in `_inductor`, but trying to introduce an intermediate class `DataNode` with no behavior but a declared `node: IRNode` led to subtle failures in some inductor checks which I couldn't track down, and with just two classes in it, it isn't too unwieldly...